Error at syncing sincing one board of one of my accounts on one of my servers

<!--
Guidelines for submitting bug reports:

* Bug reports which do not fill the complete issue template will be closed.
* Please have a look at our [FAQ](https://github.com/stefan-niedermann/nextcloud-deck/blob/master/FAQ.md)
* Please search the existing issues first, it's likely that your issue was already reported or even fixed.
* This repository is *only* for issues within the Nextcloud Deck Android app
-->

<!-- Please keep this note for other contributors -->
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

**Describe the bug**
<!-- A clear and concise description of what the bug is. -->
Since a few updates of the android desk app I can't sync one board of one of my account on one of my servers. It says it has an error at syncing.


**Expected behavior**
<!-- A clear and concise description of what you expected to happen. -->
The desk android app should sync my board like before.


**Versions**
 - Nextcloud: 19.0.1
 - Nextcloud Deck: 1.0.5
 - Nextcloud Android: 3.12.1
 - Nextcloud Android Deck: 1.9.3


**Smartphone (please complete the following information):**
- Device: Samsung Galaxy S10 (Fingerprint: Google Pixel 4)
- Android-Version: LineageOS 17.1 (Android 10.1)
- App-Store:
  - ❔ Google Play Store
  - ✔ F-Droid


**Stacktrace**
```
App Version: 1.9.3
App Version Code: 1009003
App Flavor: fdroid

Deck Server Version: 1.0.5

Files App Version Code: 30120190

---

OS Version: 4.14.113-gfe8f81568d52(6506677)
OS API Level: 29
Device: flame
Manufacturer: Google
Model (and Product): Pixel 4 (flame)

---

it.niedermann.nextcloud.deck.exceptions.TraceableException: Sorry, a wild error appeared!

⚠️ If you want to tell us about the following issue, please make sure to censor sensitive data beforehand! ⚠️

Failed to run traceable code with arguments:
{"title":"Version 1.11","boardId":2,"deletedAt":0,"lastModified":1595999512,"cards":[{"title":"Edit camera frustrum","description":"[Reference Message](https://discordapp.com/channels/441656243346735115/441658028782125076/731627891292635248)","stackId":23,"type":"plain","lastModified":1595999512,"lastEditor":null,"createdAt":1594614450,"labels":[{"title":"Enhancement","color":"399a2c","boardId":2,"cardId":89,"lastModified":1586440033,"id":9},{"title":"Level Manager","color":"ac5753","boardId":2,"cardId":89,"lastModified":1586429414,"id":19}],"assignedUsers":[{"id":72,"participant":"pilzinsel64","cardId":89,"type":null}],"attachments":null,"attachmentCount":0,"owner":{"primaryKey":"pilzinsel64","uid":"pilzinsel64","displayname":"Pilzinsel64","type":0},"order":0,"archived":false,"duedate":null,"deletedAt":0,"commentsUnread":0,"id":89,"overdue":0},{"title":"Render optimations","description":"# Todo\n- [ ] Textures not clamping\n- [ ] IA and I textures are invisible\n- [ ] vertex colorings appearing way off\n\n# References\n\n[Reference Message](https://discordapp.com/channels/441656243346735115/441657959911653376/732593604177952800)","stackId":23,"type":"plain","lastModified":1595999512,"lastEditor":null,"createdAt":1594741953,"labels":[{"title":"Bug","color":"ad1818","boardId":2,"cardId":91,"lastModified":1586440039,"id":10},{"title":"Confirmed","color":"317CCC","boardId":2,"cardId":91,"lastModified":1586428742,"id":13}],"assignedUsers":[],"attachments":null,"attachmentCount":0,"owner":{"primaryKey":"pilzinsel64","uid":"pilzinsel64","displayname":"Pilzinsel64","type":0},"order":1,"archived":false,"duedate":null,"deletedAt":0,"commentsUnread":0,"id":91,"overdue":0},{"title":"Import/Export (part of) Levels","description":"# Todo\n## General\n- [x] Prepair code for <font  color=\"green\">JSON</font> serialization\n- [ ] Implementation to Level Manager\n- [x] Controller\n\n## Functions\n- [x] Export functions\n- [x] Import functions\n\n## GUI\n- [ ] GUI for import\n\n# Infos\n## Features\n- Be able to export the whole area or just some parts of it.\n- Be able to import the whole area (if available) from that file or just parts of it.\n- What can be exported (selected, Radio Buttons):\n  - Whole area\n  - Objects\n    - Include all\n    - Choose single ones\n  - Warps\n    - Include all\n    - Choose single ones\n## File formats\n- JSON file format\n  - Extra class that contains a list of the new type `ContentHolder` that has an property `object Content`.\n  - Serialize with `@type` node (remember type of objects).\n# Feature Requests\n## TheGreatDuck#9159\nSee [📎 beeg_request.txt](/pilzcloud/apps/deck/cards/90/attachment/10)","stackId":23,"type":"plain","lastModified":1595999512,"lastEditor":null,"createdAt":1594701665,"labels":[{"title":"Enhancement","color":"399a2c","boardId":2,"cardId":90,"lastModified":1586440033,"id":9},{"title":"Confirmed","color":"317CCC","boardId":2,"cardId":90,"lastModified":1586428742,"id":13},{"title":"Doing","color":"094195","boardId":2,"cardId":90,"lastModified":1586956672,"id":16},{"title":"Level Editor","color":"ac5753","boardId":2,"cardId":90,"lastModified":1586429423,"id":20}],"assignedUsers":[{"id":73,"participant":"pilzinsel64","cardId":90,"type":null}],"attachments":null,"attachmentCount":1,"owner":{"primaryKey":"pilzinsel64","uid":"pilzinsel64","displayname":"Pilzinsel64","type":0},"order":2,"archived":false,"duedate":null,"deletedAt":0,"commentsUnread":0,"id":90,"overdue":0},{"title":"Data Editor for Objects and Warps","description":"# Todo\n- Edit Objects and Warps in a Data Table\n  - [ ] Normal 3D Objects\n  - [ ] Connected Warps\n  - [ ] Instant Warps\n  - [ ] Painting Warps\n- [ ] Import function\n- [ ] Export function\n- [ ] GUI\n\n# Features\n- Edit Objects and Warps in a Data Table.\n- Export all or selected content as CSV\n  - Include file formats to SaveFileDialog: VSC\n  - Add radiobuttons to SavFileDialog:\n    - Splitter char: `;` (Default)\n    - Splitter char: `,`\n- Import all content from a CSV\n# Feature Requests\n## TheGreatDuck#9159\nSee [📎 beeg_request.txt](/pilzcloud/apps/deck/cards/90/attachment/10)","stackId":23,"type":"plain","lastModified":1595999512,"lastEditor":null,"createdAt":1595581955,"labels":[{"title":"Enhancement","color":"399a2c","boardId":2,"cardId":96,"lastModified":1586440033,"id":9},{"title":"Confirmed","color":"317CCC","boardId":2,"cardId":96,"lastModified":1586428742,"id":13},{"title":"Doing","color":"094195","boardId":2,"cardId":96,"lastModified":1586956672,"id":16},{"title":"New Module","color":"ac5753","boardId":2,"cardId":96,"lastModified":1586429394,"id":17},{"title":"Level Editor","color":"ac5753","boardId":2,"cardId":96,"lastModified":1586429423,"id":20}],"assignedUsers":[],"attachments":null,"attachmentCount":0,"owner":{"primaryKey":"pilzinsel64","uid":"pilzinsel64","displayname":"Pilzinsel64","type":0},"order":3,"archived":false,"duedate":null,"deletedAt":0,"commentsUnread":0,"id":96,"overdue":0},{"title":"Shadow Editor","description":"# Feature Request\n```\nAnonyme Antwort\nFreitag, 24. Juli 2020 08:29\n\n1. For what module is this feature?\nCustom Objects\n\n2. How would you name your feature?\nShadow Editor\n\n3. Please describe your feature detailed:\nAllow users to add and edit shadows on imported models. Models are currently imported with hardcoded geolayouts in `Geolayout.cs` using the `0x20` instruction. Shadows can be added using the `0x16` instruction, but it is 4 bytes longer than the `0x20` instruction, so this cannot be done via hex editing. This must be done in Rom Manager to prevent overwriting data in the ROM. Users should be able to edit the shadow type, size, and opacity. I have already done some coding for this to provide wireframes, and can also complete the feature if you would like to accept my help on this.\n\n4. Please provide me some screenshots, mockups or other helpful ressources, if possible: (🗒1)\ncontext menu wireframe: https://ibb.co/dBwTfFM\ndialog wireframe https://ibb.co/PrvwyPm\nGeolayout `0x16` instruction documentation: https://hack64.net/wiki/doku.php?id=super_mario_64:geometry_layout_commands#start_geo_layout_with_shadow\n\n5. What version of the ROM Manager are you using right now? (e.g. \"Version 1.9.0.0 Beta\")\n1.9.15\n\n6. How can you be contacted If there are more questions open?\nDiscord: Zenonx#9532\n```","stackId":23,"type":"plain","lastModified":1595999512,"lastEditor":null,"createdAt":1595833403,"labels":[{"title":"Enhancement","color":"399a2c","boardId":2,"cardId":97,"lastModified":1586440033,"id":9},{"title":"Done","color":"7C31CC","boardId":2,"cardId":97,"lastModified":1586428755,"id":11},{"title":"Custom Objects Manager","color":"ac5753","boardId":2,"cardId":97,"lastModified":1586611648,"id":33}],"assignedUsers":[],"attachments":null,"attachmentCount":0,"owner":{"primaryKey":"pilzinsel64","uid":"pilzinsel64","displayname":"Pilzinsel64","type":0},"order":4,"archived":false,"duedate":null,"deletedAt":0,"commentsUnread":0,"id":97,"overdue":0}],"order":3,"id":23}
Cause: Sorry, a wild error appeared!

⚠️ If you want to tell us about the following issue, please make sure to censor sensitive data beforehand! ⚠️

Failed to run traceable code with arguments:
{"title":"Edit camera frustrum","description":"[Reference Message](https://discordapp.com/channels/441656243346735115/441658028782125076/731627891292635248)","stackId":23,"type":"plain","lastModified":1595999512,"lastEditor":null,"createdAt":1594614450,"labels":[{"title":"Enhancement","color":"399a2c","boardId":2,"cardId":89,"lastModified":1586440033,"id":9},{"title":"Level Manager","color":"ac5753","boardId":2,"cardId":89,"lastModified":1586429414,"id":19}],"assignedUsers":[{"id":72,"participant":"pilzinsel64","cardId":89,"type":null}],"attachments":null,"attachmentCount":0,"owner":{"primaryKey":"pilzinsel64","uid":"pilzinsel64","displayname":"Pilzinsel64","type":0},"order":0,"archived":false,"duedate":null,"deletedAt":0,"commentsUnread":0,"id":89,"overdue":0}
Cause: Not a JSON Object: "pilzinsel64"
	at it.niedermann.nextcloud.deck.api.JsonToEntityParser.parseStack(JsonToEntityParser.java:530)
	at it.niedermann.nextcloud.deck.api.JsonToEntityParser.parseJsonObject(JsonToEntityParser.java:51)
	at it.niedermann.nextcloud.deck.api.NextcloudArrayDeserializer.deserialize(NextcloudArrayDeserializer.java:36)
	at it.niedermann.nextcloud.deck.api.NextcloudArrayDeserializer.deserialize(NextcloudArrayDeserializer.java:18)
	at com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:69)
	at com.google.gson.Gson.fromJson(Gson.java:932)
	at com.google.gson.Gson.fromJson(Gson.java:897)
	at com.nextcloud.android.sso.api.NextcloudAPI.convertStreamToTargetEntity(NextcloudAPI.java:122)
	at com.nextcloud.android.sso.api.NextcloudAPI.lambda$performRequestObservableV2$1$NextcloudAPI(NextcloudAPI.java:98)
	at com.nextcloud.android.sso.api.-$$Lambda$NextcloudAPI$HPzOdi9LOc_f-wA9GNnhc9LKQSM.subscribe(Unknown Source:6)
	at io.reactivex.internal.operators.observable.ObservableFromPublisher.subscribeActual(ObservableFromPublisher.java:31)
	at io.reactivex.Observable.subscribe(Observable.java:12284)
	at io.reactivex.internal.operators.observable.ObservableMap.subscribeActual(ObservableMap.java:32)
	at io.reactivex.Observable.subscribe(Observable.java:12284)
	at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeTask.run(ObservableSubscribeOn.java:96)
	at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:919)
Caused by: it.niedermann.nextcloud.deck.exceptions.TraceableException: Sorry, a wild error appeared!

⚠️ If you want to tell us about the following issue, please make sure to censor sensitive data beforehand! ⚠️

Failed to run traceable code with arguments:
{"title":"Edit camera frustrum","description":"[Reference Message](https://discordapp.com/channels/441656243346735115/441658028782125076/731627891292635248)","stackId":23,"type":"plain","lastModified":1595999512,"lastEditor":null,"createdAt":1594614450,"labels":[{"title":"Enhancement","color":"399a2c","boardId":2,"cardId":89,"lastModified":1586440033,"id":9},{"title":"Level Manager","color":"ac5753","boardId":2,"cardId":89,"lastModified":1586429414,"id":19}],"assignedUsers":[{"id":72,"participant":"pilzinsel64","cardId":89,"type":null}],"attachments":null,"attachmentCount":0,"owner":{"primaryKey":"pilzinsel64","uid":"pilzinsel64","displayname":"Pilzinsel64","type":0},"order":0,"archived":false,"duedate":null,"deletedAt":0,"commentsUnread":0,"id":89,"overdue":0}
Cause: Not a JSON Object: "pilzinsel64"
	at it.niedermann.nextcloud.deck.api.JsonToEntityParser.parseCard(JsonToEntityParser.java:340)
	at it.niedermann.nextcloud.deck.api.JsonToEntityParser.lambda$parseStack$12(JsonToEntityParser.java:545)
	at it.niedermann.nextcloud.deck.api.-$$Lambda$JsonToEntityParser$PlcXUu1u8LNgp7aOPq437OOzOqY.run(Unknown Source:6)
	at it.niedermann.nextcloud.deck.exceptions.TraceableException.makeTraceableIfFails(TraceableException.java:13)
	... 23 more
Caused by: java.lang.IllegalStateException: Not a JSON Object: "pilzinsel64"
	at com.google.gson.JsonElement.getAsJsonObject(JsonElement.java:91)
	at it.niedermann.nextcloud.deck.api.JsonToEntityParser.lambda$parseCard$8(JsonToEntityParser.java:365)
	at it.niedermann.nextcloud.deck.api.-$$Lambda$JsonToEntityParser$ar3RHPP0QM1zvvkXYGeTM-aSu2Y.run(Unknown Source:6)
	at it.niedermann.nextcloud.deck.exceptions.TraceableException.makeTraceableIfFails(TraceableException.java:13)
	... 27 more

```

--- a/app/src/main/java/it/niedermann/nextcloud/deck/api/JsonToEntityParser.java
+++ b/app/src/main/java/it/niedermann/nextcloud/deck/api/JsonToEntityParser.java
@@ -303,7 +303,7 @@ public class JsonToEntityParser {
                 if (owner.isJsonPrimitive()) {//TODO: remove if, let only else!
                     DeckLog.verbose("owner is Primitive, skipping");
                 } else
-                    fullBoard.setOwner(parseUser(owner.getAsJsonObject()));
+                    fullBoard.setOwner(parseUser(owner));
             }
         }, e);
         return fullBoard;
@@ -315,7 +315,7 @@ public class JsonToEntityParser {
 
         if (aclJson.has("participant") && !aclJson.get("participant").isJsonNull()) {
             makeTraceableIfFails(() -> {
-                User participant = parseUser(aclJson.get("participant").getAsJsonObject());
+                User participant = parseUser(aclJson.get("participant"));
                 acl.setUser(participant);
                 acl.setType(aclJson.get("type").getAsLong());
                 acl.setBoardId(aclJson.get("boardId").getAsLong());
@@ -362,7 +362,7 @@ public class JsonToEntityParser {
                 for (JsonElement assignedUser : assignedUsers) {
                     JsonObject userJson = assignedUser.getAsJsonObject();
                     if (userJson.has("participant") && !userJson.get("participant").isJsonNull()) {
-                        users.add(parseUser(userJson.get("participant").getAsJsonObject()));
+                        users.add(parseUser(userJson.get("participant")));
                     }
                 }
                 fullCard.setAssignedUsers(users);
@@ -388,10 +388,7 @@ public class JsonToEntityParser {
             card.setCommentsUnread(e.get("commentsUnread").getAsInt());
             JsonElement owner = e.get("owner");
             if (owner != null) {
-                if (owner.isJsonPrimitive()) {//TODO: remove if, let only else!
-                    DeckLog.verbose("owner is Primitive, skipping");
-                } else
-                    fullCard.setOwner(parseUser(owner.getAsJsonObject()));
+                fullCard.setOwner(parseUser(owner));
             }
             card.setArchived(e.get("archived").getAsBoolean());
         }, e);
@@ -430,14 +427,27 @@ public class JsonToEntityParser {
         return a;
     }
 
-    protected static User parseUser(JsonObject e) {
-        DeckLog.verbose(e.toString());
+    protected static User parseUser(JsonElement userElement) {
+        DeckLog.verbose(userElement.toString());
+        if (userElement.isJsonNull()){
+            return null;
+        }
         User user = new User();
         makeTraceableIfFails(() -> {
-            user.setDisplayname(getNullAsEmptyString(e.get("displayname")));
-            user.setPrimaryKey(getNullAsEmptyString(e.get("primaryKey")));
-            user.setUid(getNullAsEmptyString(e.get("uid")));
-        }, e);
+
+            if (userElement.isJsonPrimitive()) {
+                String uid = userElement.getAsString();
+                user.setDisplayname(uid);
+                user.setPrimaryKey(uid);
+                user.setUid(uid);
+            } else {
+                JsonObject userJson = userElement.getAsJsonObject();
+                user.setDisplayname(getNullAsEmptyString(userJson.get("displayname")));
+                user.setPrimaryKey(getNullAsEmptyString(userJson.get("primaryKey")));
+                user.setUid(getNullAsEmptyString(userJson.get("uid")));
+            }
+
+        }, userElement);
         return user;
     }
 

